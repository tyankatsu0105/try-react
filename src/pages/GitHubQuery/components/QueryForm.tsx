import React, { FunctionComponent } from 'react';

export const QueryForm: FunctionComponent = () => {
  return (
    <>
      <a href="">aaaaa</a>
    </>
  );
};
