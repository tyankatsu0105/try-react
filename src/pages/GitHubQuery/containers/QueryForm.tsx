import React, { FunctionComponent } from 'react';

import { useQuery } from '@apollo/react-hooks';
import gqlGetOrganization from './api/GetOrganization.graphql';
import {
  GetOrganization,
  GetOrganizationVariables,
} from './api/__generated__/GetOrganization';

import { useFormik } from 'formik';
import * as Yup from 'yup';
import {
  FormControl,
  InputLabel,
  Input,
  FormHelperText,
  FormGroup,
  Button,
  Box,
  Typography,
  CircularProgress,
  Divider,
  Link,
} from '@material-ui/core';

interface FormValues {
  login: string;
  first: number;
}

// import { QueryFormComponent } from '../components';

export const QueryForm: FunctionComponent = () => {
  const variables: GetOrganizationVariables = {
    login: '',
    first: 0,
  };
  const { loading, data, refetch } = useQuery<GetOrganization>(
    gqlGetOrganization,
    {
      variables,
    }
  );

  const { values, errors, touched, handleChange, handleSubmit } = useFormik<
    FormValues
  >({
    initialValues: {
      login: '',
      first: 1,
    },
    validationSchema: Yup.object().shape({
      login: Yup.string().required('入力必須'),
      first: Yup.number()
        .min(1)
        .required('入力必須'),
    }),
    onSubmit: (values) => {
      refetch(values);
    },
  });
  return (
    <>
      {/* <QueryFormComponent /> */}
      <form action="" onSubmit={handleSubmit} autoComplete="off">
        <FormGroup>
          <Box marginTop={2}>
            <FormControl error={touched.login && errors.login ? true : false}>
              <InputLabel htmlFor="login">organization</InputLabel>
              <Input
                type="input"
                name="login"
                value={values.login}
                id="login"
                aria-describedby="input-login"
                onChange={handleChange}
              />

              <FormHelperText id="login">
                {touched.login && errors.login
                  ? errors.login
                  : 'organization名を入力'}
              </FormHelperText>
            </FormControl>
          </Box>

          <Box marginTop={2}>
            <FormControl error={touched.first && errors.first ? true : false}>
              <InputLabel htmlFor="first">organization</InputLabel>
              <Input
                type="number"
                name="first"
                value={values.first}
                id="first"
                aria-describedby="input-first"
                onChange={handleChange}
              />

              <FormHelperText id="first">
                {touched.first && errors.first
                  ? errors.first
                  : '表示するレポジトリ数を入力'}
              </FormHelperText>
            </FormControl>
          </Box>

          <Box marginTop={2} display="flex" justifyContent="center">
            <Button type="submit" variant="contained" color="primary">
              submit
            </Button>
          </Box>
        </FormGroup>
      </form>

      {loading && (
        <>
          <CircularProgress></CircularProgress>
        </>
      )}

      {data &&
        data.organization &&
        data.organization.repositories &&
        data.organization.repositories.nodes && (
          <>
            {data.organization.repositories.nodes.map((node) => {
              return (
                node && (
                  <>
                    <Box marginTop={1}>
                      <Link href={node.url}>
                        <Typography variant="h2">{node.name}</Typography>
                      </Link>
                      <Typography paragraph={true}>
                        {node.description}
                      </Typography>
                    </Box>
                    <Divider />
                  </>
                )
              );
            })}
          </>
        )}
    </>
  );
};
