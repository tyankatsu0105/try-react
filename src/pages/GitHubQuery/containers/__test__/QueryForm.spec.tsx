import React from 'react';

import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import { MockedProvider } from '@apollo/react-testing';
import gqlGetOrganization from '../api/GetOrganization.graphql';

import { QueryForm } from '../QueryForm';

const mocks = [
  {
    request: {
      query: gqlGetOrganization,
      variables: {
        login: 'facebook',
        first: 1,
      },
    },
    result: {
      data: {
        organization: {
          repositories: {
            nodes: [
              {
                name: 'codemod',
                description:
                  'Codemod is a tool/library to assist you with large-scale codebase refactors that can be partially automated but still require human oversight and occasional intervention. Codemod was developed at Facebook and released as open source.',
                url: 'https://github.com/facebook/codemod',
              },
            ],
          },
        },
      },
    },
  },
];

describe('QueryForm', () => {
  it('render', () => {
    const { baseElement } = render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <QueryForm />
      </MockedProvider>
    );

    expect(baseElement).toMatchSnapshot();
  });
});
