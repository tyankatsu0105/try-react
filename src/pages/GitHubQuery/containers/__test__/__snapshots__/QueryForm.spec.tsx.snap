// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryForm render 1`] = `
<body>
  <div>
    <form
      action=""
      autocomplete="off"
    >
      <div
        class="MuiFormGroup-root"
      >
        <div
          class="MuiBox-root MuiBox-root-4"
        >
          <div
            class="MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink="false"
              for="login"
            >
              organization
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            >
              <input
                aria-describedby="input-login"
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                id="login"
                name="login"
                type="input"
                value=""
              />
            </div>
            <p
              class="MuiFormHelperText-root"
              id="login"
            >
              organization名を入力
            </p>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-66"
        >
          <div
            class="MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
              data-shrink="true"
              for="first"
            >
              organization
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            >
              <input
                aria-describedby="input-first"
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                id="first"
                name="first"
                type="number"
                value="1"
              />
            </div>
            <p
              class="MuiFormHelperText-root MuiFormHelperText-filled"
              id="first"
            >
              表示するレポジトリ数を入力
            </p>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-67"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              submit
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </form>
    <div
      class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </div>
  </div>
</body>
`;
